docs/features/incremental-generators.cookbook.md

```diff
@@ -0,0 +1,654 @@
+# Incremental Generators Cookbook
+
+## Summary
+
+This document aims to be a guide to help the creation of source generators by providing a series of guidelines for common patterns.
```

Comment:
i have only skimmed.  Do we have anywhere where we strongly recommend that generators just produce a string simply by appending to a string builder?  We should heavily push people away from more complex systems, *Especially* ones that use Syntax as an intermediary representation. 

For context, there is an internal discussion with an internal team, where using a complex building system is adding dozens of seconds to their generation costs.  Note: we could even link to refsergio's indenting string builder type as a useful helper here.

Reply:
[Here it is](https://github.com/Sergio0694/ComputeSharp/blob/main/src/ComputeSharp.SourceGeneration/Helpers/IndentedTextWriter.cs), if it helps 🙂

Also includes the new interpolated string handler goodness for maximum perf and usability. Though worth noting that that requires at least C# 10 to work, as well as some polyfill types, so it kinda ties back to the other conversation (which we might want to continue offline first) about also considering to officially recommend PolySharp too, so generator authors can also get new C# and can use stuff like this.